include/fmt/ostream.h

```diff
@@ -38,24 +38,24 @@ auto get_file(std::filebuf&) -> FILE*;
 #endif
 
 inline bool write_ostream_unicode(std::ostream& os, fmt::string_view data) {
+  FILE* c_file = nullptr;
 #if FMT_MSC_VERSION
   if (auto* buf = dynamic_cast<std::filebuf*>(os.rdbuf()))
-    if (FILE* f = get_file(*buf)) return write_console(f, data);
+    c_file = get_file(*buf);
 #elif defined(_WIN32) && defined(__GLIBCXX__)
   auto* rdbuf = os.rdbuf();
-  FILE* c_file;
   if (auto* sfbuf = dynamic_cast<__gnu_cxx::stdio_sync_filebuf<char>*>(rdbuf))
     c_file = sfbuf->file();
   else if (auto* fbuf = dynamic_cast<__gnu_cxx::stdio_filebuf<char>*>(rdbuf))
     c_file = fbuf->file();
   else
     return false;
-  if (c_file) return write_console(c_file, data);
-#else
-  ignore_unused(os, data);
 #endif
-  return false;
+  if (!c_file) return false;
+  os.flush();
```

Comment:
We should only flush the buffer when writing to a console. I suggest moving the ostream change to a separate PR and addressing this issue. The rest LGTM.